### DISCUSSION TITLE
Unable to get logging working

### DISCUSSION THREAD

**Discussion by jwaltrip4** [abraunegg/onedrive#1772] [General]
Set up .config/onedrive/config to point to /var/log/onedrive
created /var/log/onedrive folder
--enable-logging set

Still does not write the log.   What am I missing :)

**Comment by abraunegg** [comment]
@jwaltrip4 
> Still does not write the log. What am I missing

As you are using Ubuntu based on your prior issue - how did you 'upgrade' your client - via manual install or via installing from the OpenSuSE Build Service (OBS) ?

If you did not read the following, this is the first issue you are hitting:
* https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages
* https://github.com/abraunegg/onedrive/issues/1274

You need to ensure you 'fix' this problem first - or dont use Ubuntu / Ubuntu clones. You need to reboot your system to ensure that the 'systemd' service that the Ubuntu package process creates has been removed and not started.

Once that is sorted & confirmed to be fixed you can look at your logging issue. To do this:
1. Stop any systemd service that might be configured / running
2. Run the client to see what configuration options are actually configured | set: `onedrive --display-config`. Make sure that all these options are what you have configured
3. Test the client manually in verbose mode with : `onedrive --synchronize --dry-run --verbose` .. what is the client telling you in regards to configuring logging and being able to use the logging directory? 
4. Repeat 2 and 3 until the client is using logging directory
5. Run the client manually to confirm the use of the logging directory: `onedrive --synchronize --verbose`

**Important Additional Note:**
Additionally in your initial issue, you were running the client via 'sudo onedrive'. Do not run the client in this way.

When you do this, you are running the client as 'root', using the configuration as root, thus all your files & data will be permission as 'root'. This will lead to access issues & configuration issues - example - you have configured logging under your 'normal account' (thus `~/.config/onedrive/config` - but if you are using 'sudo onedrive' to run, your configuration is coming from `/root/.config/onedrive/config` ...

You do not need to run the client with 'sudo' for correct operation.

Last note - in the future please be a little more verbose when raising questions ...

**Reply by jwaltrip4** [reply]
thanks for the quick reply....  I have removed the symlink.  I also checked to make sure OneDrive is not running as a service.

I have my config file both in ~/.config/onedrive and in /etc/onedrive.  I assume not necessary.  What is the best spot?

Here is my Config file:
```
sync_dir = "~/James_OD"
skip_dir = "Apps|Attachments|Documents|Photos|videos|Screenshots|Download"
skip_dir = "Samsung Gallery"
skip_dir = "DCIM"
log_dir = "/var/log/onedrive/"
enable_logging = "true"
monitor_log_frequency = "5"
sync_dir_permissions = "755"
sync_file_permissions = "755"**
```

Here is the results of Display-Config
```
Configuration file successfully loaded
onedrive version                       = v2.4.14-1~np1
Config path                            = /home/jwaltrip/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/jwaltrip/James_OD
Config option 'skip_dir'               = Apps|Attachments|Documents|Photos|videos|Screenshots|Download|Samsung Gallery|DCIM
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false

```
Should I be seeing all of the stuff I set in config, under display config?

**Reply by abraunegg** [reply]
You are seeing all of the correct items as per application output for `--display-config`

What now happens when you run the client in verbose mode with dry-run? what does the client tell you regarding logging?

**Reply by abraunegg** [reply]
> I have my config file both in ~/.config/onedrive and in /etc/onedrive. I assume not necessary. What is the best spot?

The application *will* only look in `~/.config/onedrive/` for the configuration file unless you set & use the `--confdir` parameter each and every time to specify an alternate configuration file.

**Reply by jwaltrip4** [reply]
Here is results of dry run:

Still is not writing log to /var/log/onedrive

```
jwaltrip@plex-media-svr:~$ onedrive --synchronize --dry-run --verbose
Using 'user' Config Dir: /home/jwaltrip/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
Configuration file successfully loaded
DRY-RUN Configured. Output below shows what 'would' have occurred.
Using logfile dir: /var/log/onedrive/
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Opening the item database ...
All operations will be performed in: /home/jwaltrip/James_OD
Application version: v2.4.14-1~np1
Account Type: personal
Default Drive ID: f19acb22e4c0a9ea
Default Root ID: F19ACB22E4C0A9EA!103
Remaining Free Space: 1104767485379
Fetching details for OneDrive Root
OneDrive Root exists in the database
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Applying changes of Path ID: F19ACB22E4C0A9EA!103
Updated Remaining Free Space: 1104767485379
Uploading differences of ~/James_OD
Processing .
The directory has not changed
Processing Pictures
The directory has not changed
Processing Pictures/Camera Roll
The directory has not changed
Processing Pictures/Camera Roll/2021
The directory has not changed
Processing Pictures/Camera Roll/2021/12
The directory has not changed
Processing Pictures/Camera Roll/2021/12/20211231_135900.jpg
The file has not changed
Processing Pictures/Camera Roll/2021/12/20211231_135858.jpg
The file has not changed
Processing Pictures/Camera Roll/2021/12/20211231_153355.jpg
The file has not changed
Processing Pictures/Camera Roll/2021/12/20211231_173159.jpg
The file has not changed
Processing Pictures/Camera Roll/2021/12/20211231_192251.mp4
The file has not changed
Uploading new items of ~/James_OD
Applying changes of Path ID: F19ACB22E4C0A9EA!103
Updated Remaining Free Space: 1104767485379
```

**Comment by abraunegg** [comment]
@jwaltrip4 
So this tells me your directories & access is correct:
```
jwaltrip@plex-media-svr:~$ onedrive --synchronize --dry-run --verbose
Using 'user' Config Dir: /home/jwaltrip/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
Configuration file successfully loaded
DRY-RUN Configured. Output below shows what 'would' have occurred.
Using logfile dir: /var/log/onedrive/
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
...
```

If this was not correct, the following would be logged:
```
...
Configuration file successfully loaded
config file has been updated, checking if --resync needed
DRY-RUN Configured. Output below shows what 'would' have occurred.
Using logfile dir: /var/log/onedrive/

Unable to access /var/log/onedrive/
Please manually create '/var/log/onedrive/' and set appropriate permissions to allow write access
The requested client activity log will instead be located in the users home directory

Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
```

So this means that the client is seeing the directory which is a good thing.

Have you set the correct permissions for your logging directory as per: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#client-activity-log

Without doing this step, you cannot write to the directory as your normal user.

**Reply by jwaltrip4** [reply]
Permissions are set as specified in the link.  Still no joy :)

**Comment by abraunegg** [comment]
@jwaltrip4 
Is your 'user' part of the 'users' group ?

```
cat /etc/group | grep users
```

If not:
```
sudo usermod -a -G users <your-user-name>
```
**Example:**
```
sudo usermod -a -G users alex
```

You then need to 'logout' of all sessions / SSH sessions to login again to have the new group access applied.

**Reply by jwaltrip4** [reply]
that was it.  thanks!!

**Comment by jwaltrip4** [comment]
So... as a side note.  here is what I am fooling around with.
My wife an I purchased higher end smartphones among other reasons for the excellent cameras on them.
When we take photos, We copy them to my server, and then they are backed up to other drives and the cloud.
When we get home we sort through the photos and put them in folders.  Some get posted on Facebook.  Some I put on my website.

One of the things I wanted to do...  instead of wafting to get home to copy them off, I have set up OneDrive to backup up all pictures and videos taken to my cameras Roll.  Your tool comes in at this point to copy the camera roll to an intermediate folder on my server.  I then have an rsync job that copies the files to a user accessible share.  this share is also backup up every few hours to the cloud.

thanks for all of your help.
